### ISSUE TITLE
use with ocaml 5

### ISSUE THREAD

**Issue by joprice** [ml-in-barcelona/server-reason-react#21]
What would it take to build the project on ocaml 5? I tried building from nix with an ocaml 5 overlay and ended up with a ppxlib error:

```
File "packages/intf_of_sign/intf_of_sign.ml", line 85, characters 4-26:
85 |     of_type_desc type_desc
         ^^^^^^^^^^^^^^^^^^^^^^
Error: This expression has type
         Ppxlib_ast.Ast.expression = Astlib__Ast_414.Parsetree.expression
       but an expression was expected of type Parsetree.expression
```

There is a constraint on < ocaml 5, so I assume there's some compatibility issues that need to be resolved first?

**Comment by davesnx** [comment]
Hey @joprice, currently it's mean to be used with 4.14 just because the version of ppxlib (0.23) isn't available for OCaml 5.

There's another inconvenience for OCaml 5, which isn't a blocker per se but I'm using Lwt which isn't the greatest with 5 https://github.com/ocsigen/lwt/issues/897, probably can fix it by using https://github.com/talex5/lwt_eio but still I would rather update to use eio/OCaml 5 in one go.

Regardles, I will keep the issue open.

**Comment by davesnx** [comment]
Hey @joprice, made a PR where I have building with opam and OCaml 5.1.0-rc1 and will merge soon.

**Comment by davesnx** [comment]
Merged, updating the CI to run 5.0 as well. Let me know if it works with Nix so I can close this issue!

**Comment by joprice** [comment]
@davesnx thanks! I’ve been on vacation but will try it out soon.
